### DISCUSSION TITLE
TypeError: Cannot read properties of undefined (reading 'booted')

### DISCUSSION THREAD

**Discussion by equeraltowenea** [adonisjs/core#4675] [Help]
When I move the models to a local library, I get the following error when trying to use it for the Auth Guard.

```
TypeError: Cannot read properties of undefined (reading 'booted')
 

   ⁃ importFile
     node_modules\@poppinss\utils\src\fs_import_all.ts:48
   ⁃ fsImportAll
     node_modules\@poppinss\utils\src\fs_import_all.ts:90
   ⁃ ConfigManager.process
     node_modules\@adonisjs\application\src\managers\config.ts:54
```

Project estructure:

- Project
  - app1
  - common

app1 package.json:

```
"dependencies": {
  ...
  "@common": "file:../common",
  ...
}

```

**Comment by RomainLanz** [comment]
Hey @equeraltovivas! 👋🏻 

You need to ensure you are using exactly the same version of Lucid between your application and your model.

**Comment by equeraltowenea** [comment]
Hi @RomainLanz ,
Adding exactly same dependencies with same versions still produces error. It doesn't seem to like that I have those models in a shared folder outside of Adonis project.

**Comment by equeraltowenea** [comment]
I think the problem is that Lucid's BaseModel has adapter to database as a singleton property. This causes that when we move models to a shared folder between several projects it does not have adapter and it fails.

How can we keep models in a shared folder between Adonis projects?

**Comment by DavideCarvalho** [comment]
Hey @equeraltovivas have you solved it? I'm having the exact same problem

I'm using turborepo with pnpm, and I created a db package to share my models between my 2 projects

**Comment by DavideCarvalho** [comment]
To anyone that comes here in the future. Make sure you have the same adonisjs related dependencies across all your apps.

After some time debugging, PNPM joins dependencies based on their versions

So like, if one app has adonis core@6.x and ts at @5.2, and another app has the same version of adonis core, but ts is idk, 5.5, it will bundle 2 adonis core

What I did to fix was I created a pnpm catalog with all adonisjs related dependencies. This is my catalog, I hope is helps

catalogs:
  adonis:
    "@adonisjs/ally": "5.1.1"
    "@adonisjs/auth": "9.5.0"
    "@adonisjs/bouncer": "3.1.6"
    "@adonisjs/core": "6.19.0"
    "@adonisjs/cors": "2.2.1"
    "@adonisjs/drive": "3.4.1"
    "@adonisjs/i18n": "2.2.2"
    "@adonisjs/inertia": "3.1.1"
    "@adonisjs/lock": "1.1.1"
    "@adonisjs/lucid": "21.8.0"
    "@adonisjs/lucid-slugify": "3.0.0"
    "@adonisjs/mail": "9.2.2"
    "@adonisjs/redis": "9.2.0"
    "@adonisjs/session": "7.5.1"
    "@adonisjs/shield": "8.2.0"
    "@adonisjs/static": "1.1.1"
    "@adonisjs/vite": "4.0.0"
    "@adonisjs/assembler": "7.8.2"
    "@adonisjs/eslint-config": "2.1.2"
    "@adonisjs/prettier-config": "1.4.5"
    "@adonisjs/tsconfig": "1.4.1"
    "@tuyau/client": "0.2.10"
    "@tuyau/core": "0.4.2"
    "@tuyau/inertia": "0.0.15"
    "@tuyau/superjson": "0.1.2"
    "@tuyau/utils": "0.0.9"
    "typescript": "5.9.3"
    "@types/node": "24.6.1"
    "@types/luxon": "3.7.1"
    "@types/react": "19.2.0"
    "@types/react-dom": "19.2.0"
    "@types/tdigest": "0.1.5"
